### PR TITLE
Assigning categories for Task101-121

### PR DESCRIPTION
**Pull request by garyhlai**
- update domains & categories for task101-121

### CODE REVIEW

File: doc/task-hierarchy.md [hunk 1]
```diff
@@ -140,7 +139,10 @@
   - `Text Generation -> Weblink Generation`
 - `Text Modification`
   - `Text Modification -> Structured Text Modification`
-  - `Text Modification -> Text Simplification`
+    - `Text Modification -> Structured Text Modification -> Logical Operation`: An unambiguous transform is applied to all inputs e.g. subtract 1 from every element in the list
+  - `Text Modification -> Style Transfer`
+    `Text Modification -> Style Transfer -> Text Simplification`
+  - `Text Modification -> Paraphrasing`
 - `Text Span Selection`
```
Comment (garyhlai) [inline doc/task-hierarchy.md:167]: Structured Text Processing -> Semantic Parsing -> Program Synthesis
Structured Text Processing -> Code Summarization
Structured Text Processing -> Operation on Primitives -> Numbers: e.g. An unambiguous transform is applied to all inputs e.g. subtract 1 from every element in the list
     Structured Text Processing -> Operation on Primitives -> Numbers -> List
     Structured Text Processing -> Operation on Primitives -> Numbers -> Set
     Structured Text Processing -> Operation on Primitives -> Numbers -> Scalar
Structured Text Processing -> Operation on Primitives -> String

File: doc/task-hierarchy.md [hunk 2]
```diff
@@ -112,8 +110,6 @@
 - `Sentiment Analysis`
 - `Sorting`
 - `Stance Detection`
-- `Story Completion`
```
Comment (yeganehkordi) [inline doc/task-hierarchy.md:121]: `Story Completion` is not a redundant category. Some tasks are to choose a part of a story from the options. So, we can't use text generation there. Maybe we can drop `Text Generation -> Sentence Generation -> Story Completion` and use both `Text Generation` and `Story Completion` in the generation tasks.

File: doc/task-hierarchy.md [hunk 3]
```diff
@@ -90,14 +88,14 @@
   - `Reasoning -> Deductive Reasoning`
   - `Reasoning -> Discrete Reasoning`
   - `Reasoning -> Logical Reasoning`
+    - `Reasoning -> Logical Reasoning -> Reasoning with Symbols`: Tasks where symbols represent various things e.g. if X is the number of apples in the fridge today morning and Y is the number remaining after I ate a few apples, X-Y is the number of apples I ate.
```
Comment (yeganehkordi) [inline doc/task-hierarchy.md:97]: Is the reasoning with symbols always logical reasoning?

File: doc/task-hierarchy.md [hunk 4]
```diff
@@ -167,6 +169,7 @@
     * `Code -> Repo -> Github`
     * `Code -> Repo -> Stack Overflow`
 * `Commonsense`
+  * `Commonsense -> Image Caption`
```
Comment (yeganehkordi) [inline doc/task-hierarchy.md:193]: I guess it's a rare category. Do we have more than five image caption tasks?

File: tasks/task093_conala_normalize_lists.json [hunk 5]
```diff
@@ -8,7 +8,8 @@
     "Categories": [
         "Question Answering -> Numerical Question Answering",
```
Comment (garyhlai) [inline tasks/task093_conala_normalize_lists.json]: numerical QA?

File: doc/task-hierarchy.md (hunk 4 above)
Comment (garyhlai) [reply doc/task-hierarchy.md:193]: yes

File: tasks/task105_story_cloze-rocstories_sentence_generation.json [hunk 6]
```diff
@@ -6,7 +6,12 @@
         "Story Cloze and ROCStories (https://cs.rochester.edu/nlp/rocstories/)"
     ],
     "Categories": [
-        "Story Completion"
+        "Text Generation -> Sentence Generation -> Story Completion",
+        "Reasoning -> Commonsense Reasoning"
+    ],
+    "Domains": [
+        "Story",
+        "Commonsense"
```
Comment (garyhlai) [inline tasks/task105_story_cloze-rocstories_sentence_generation.json:14]: story and commonsense?

File: tasks/task110_logic2text_sentence_generation.json [hunk 7]
```diff
@@ -6,7 +6,11 @@
         "Logic2Text"
     ],
     "Categories": [
-        "Sentence Generation"
+        "Text Generation -> Sentence Generation -> Code Summarization",
+        "Reasoning -> Logical Reasoning -> Reasoning with Symbols"
```
Comment (garyhlai) [inline tasks/task110_logic2text_sentence_generation.json:10]: reasoning with symbols?

File: tasks/task113_count_frequency_of_letter.json [hunk 8]
```diff
@@ -6,7 +6,12 @@
         "Synthetic"
     ],
     "Categories": [
-        "Answer Generation"
+        "Mathematics -> Counting",
+        "Reasoning -> Numerical Reasoning",
```
Comment (garyhlai) [inline tasks/task113_count_frequency_of_letter.json:10]: numerical reasoning?

File: tasks/task115_help_advice_classification.json [hunk 9]
```diff
@@ -3,11 +3,14 @@
...
     ],
     "Source": [
-        "Help dataset from EMNLP 2020 paper: Help! Need Advice on Identifying Advice"
+        "Help! Need Advice on Identifying Advice (https://aclanthology.org/2020.emnlp-main.427.pdf)"
     ],
     "Categories": [
         "Classification"
     ],
+    "Domains": [
+        "Dialogue"
```
Comment (garyhlai) [inline tasks/task115_help_advice_classification.json:12]: domains

File: tasks/task119_semeval_2019_task10_geometric_mathematical_answer_generation.json [hunk 10]
```diff
@@ -6,7 +6,13 @@
         "semeval 2019 task10"
     ],
     "Categories": [
-        "Math Question Answering"
+        "Question Answering -> Numerical Question Answering",
```
Comment (garyhlai) [inline tasks/task119_semeval_2019_task10_geometric_mathematical_answer_generation.json:9]: numerical

File: tasks/task120_zest_text_modification.json [hunk 11]
```diff
@@ -6,15 +6,21 @@
         "zest (https://github.com/allenai/zest)"
     ],
     "Categories": [
-        "Question Generation"
+        "Question Generation -> Contextual Question Generation"
```
Comment (garyhlai) [inline tasks/task120_zest_text_modification.json]: text modification

File: tasks/task093_conala_normalize_lists.json (hunk 5 above)
Comment (garyhlai) [reply tasks/task093_conala_normalize_lists.json]: remove

File: tasks/task115_help_advice_classification.json (hunk 9 above)
Comment (garyhlai) [reply tasks/task115_help_advice_classification.json:12]: Social Media -> Reddit

File: doc/task-hierarchy.md (hunk 2 above)
Comment (garyhlai) [reply doc/task-hierarchy.md:121]: ok let's keep both

File: doc/task-hierarchy.md (hunk 3 above)
Comment (garyhlai) [reply doc/task-hierarchy.md:97]: keeping both

**Review by swarooprm — APPROVED** [review]
LGTM, as discussed in the meeting today.

**Pull request merged**